File: domain-driven-design/python/domain/order.py
# domain/order.py
class Order:
    def __init__(self, id, customer_name, amount):
        self.id = id
        self.customer_name = customer_name
        self.amount = amount
        self.status = 'CREATED'

    def cancel(self):
        if self.status == 'CREATED':
            self.status = 'CANCELED'
            print(f"订单 ID {self.id} 已取消")
        else:
            print(f"订单 ID {self.id} 已经取消，无法重复操作")

    def display(self):
        print(f"订单 ID: {self.id}")
        print(f"客户名称: {self.customer_name}")
        print(f"订单金额: {self.amount:.2f}")
        print(f"订单状态: {self.status}")

File: domain-driven-design/rust/domain/order.rs
// domain/order.rs
// domain/order.rs
#[derive(Debug, Clone)] // 添加 Clone 特性
pub struct Order {
    id: u32,
    customer_name: String,
    amount: f64,
    status: String,
}

impl Order {
    pub fn new(id: u32, customer_name: String, amount: f64) -> Self {
        Order {
            id,
            customer_name,
            amount,
            status: String::from("CREATED"),
        }
    }

    pub fn cancel(&mut self) {
        if self.status == "CREATED" {
            self.status = String::from("CANCELED");
            println!("订单 ID {} 已取消", self.id);
        } else {
            println!("订单 ID {} 已经取消，无法重复操作", self.id);
        }
    }

    pub fn display(&self) {
        println!("订单 ID: {}", self.id);
        println!("客户名称: {}", self.customer_name);
        println!("订单金额: {:.2}", self.amount);
        println!("订单状态: {}", self.status);
    }

    pub fn get_id(&self) -> u32 {
        self.id
    }
}